### PR TITLE
Fix(Bounty-Page): Enhanced Navigation for Direct Access to Bounty Pages

### PR DESCRIPTION
**Pull request by MirzaHanan**
### Problem:
Users experienced inconsistent navigation behavior when closing a bounty page. Specifically, direct access to a bounty page (pasted URL in a new tab) and closing it redirected the user to the Google home page, instead of the more intuitive Sphinx Chat bounties homepage.

### Expected Behavior:
- When a user accesses a bounty page directly and closes it, they should be redirected to the Sphinx Chat bounties homepage (`/bounties`).
- If a user accesses a bounty page from an internal page (like their profile), closing the page should return them to the previous page, not the bounties homepage.

## Issue ticket number and link:
- **Ticket Number:** [ 1362 ]
- **Link:** [ https://github.com/stakwork/sphinx-tribes/issues/1362 ]

### Solution:
To address this issue, we implemented a conditional redirection strategy based on the user's navigation path to the bounty page. This was achieved by utilizing the `useLocation` and `useHistory` hooks from `react-router-dom` and creating a custom logic to determine the user's entry point to the bounty page.

### Changes:
- Utilized `useLocation` and `useHistory` from `react-router-dom` to manage navigation.
- Created `isDirectAccess` function to determine if the page was accessed directly (without a referrer) and contains '/bounty/' in the pathname.
- Developed `handleClose` function to apply conditional redirection: `history.push('/bounties')` for direct access and `history.goBack()` for internal navigation.
- Attached `handleClose` to relevant event handlers (`overlayClick` and `onClick` for the close button) to trigger this navigation logic.

### Evidence:
 Please see the attached video as evidence.
- Demo: [Link ](https://www.loom.com/share/7c82aeadc06f45d2bf2ee29489154bbf)

### Testing:
Thorough testing was conducted in various scenarios to ensure the functionality works as expected. This includes:
- Direct access to bounty pages and closing them.
- Accessing bounty pages from internal pages and then closing them.
- Testing in different browsers to ensure consistent behavior.

### CODE REVIEW

**Comment by MirzaHanan** [comment]
Hi @ecurrencyhodler,
Please review this PR.

**Comment by MirzaHanan** [comment]
Hi @ecurrencyhodler,
Please workflows approved.

**Comment by elraphty** [comment]
Nice job @MirzaHanan

File: frontend/app/src/components/common/Modal.tsx [hunk 1]
```diff
@@ -138,14 +141,27 @@ export default function Modal(props: ModalProps) {
       }
     : {};
 
+  const isDirectAccess = useCallback(
```
[the remark this answers is not in the document]
Comment (MirzaHanan) [reply frontend/app/src/components/common/Modal.tsx:144]: Hi @elraphty,
Thanks for highlighting this. I'll adjust the logic so it only affects the relevant modals and doesn't interfere with others like the add/edit organization modal. I'll ensure everything works smoothly and will keep you updated on the progress.

**Comment by ecurrencyhodler** [comment]
Hi! can you resubmit this PR to this new frontend repo? https://github.com/stakwork/sphinx-tribes-frontend

**Comment by MirzaHanan** [comment]
Hi @ecurrencyhodler,
Please ensure that the workflows are approved. If possible, allow the workflows to run automatically without prompting you repeatedly.

**Pull request closed without merging**